src/dmqproto/client/request/internal/Consume.d

```diff
@@ -273,60 +270,31 @@ private scope class ConsumeHandler
 
     /***************************************************************************
 
-        HandleSupportedCode policy, called from SuspendableRequestInitialiser
-        template to decide how to
-         - handle the supported code received from the node and
-         - receive the status code from the node and handle it.
-
-        Params:
-            code = supported code received from the node in response to the
-                initial message
-
-        Returns:
-            true to continue handling the request (supported and Started
-            status); false to abort (unsupported or Error status).
+        Handler policy, called from AllNodesRequest template to run the
+        request's main handling logic.
 
     ***************************************************************************/
 
-    private bool handleSupportedCode ( ubyte code )
+    private void handle ( )
     {
-        if ( !Consume.handleSupportedCodes(cast(SupportedStatus)code,
-            this.context, this.conn.remote_address) )
-        {
-            return false; // Request/version not supported
-        }
-
         // Handle initial started/error message from node.
-        auto consume_status = conn.receiveValue!(RequestStatusCode)();
-        switch (consume_status)
+        switch (conn.receiveValue!(RequestStatusCode)())
         {
-            case consume_status.Started:
+            case RequestStatusCode.Started:
```

Comment:
The general recommendation is to merge old-style "status codes" into the `MessageType` enum. One less enum = simpler. This is a protocol change, though, so we might want to consider doing this as a new request version. I'd say we can leave this as it is for now.